Add new Insteon i3 4 button KeypadLinc device

teharris1 opened the pull request:
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add support for the new Insteon i3 4 button KeypadLinc device.

Change log: https://github.com/pyinsteon/pyinsteon/compare/1.4.3...1.5.0

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] I have followed the [perfect PR recommendations][perfect-pr]
- [ ] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

joostlek committed:
Add entity translations to Mill (#96541)

gjohansson-ST committed:
Revert "Sonos add yaml config issue" (#99379)

Revert "Sonos add yaml config issue (#97365)"

This reverts commit 2299430dbeb470ff8b5a62fae1fa80fbfc3f014f.

emontnemery committed:
Add documentation URL for homeassistant_sky_connect (#99377)

AustinBrunkhorst committed:
Update pysnooz to 0.8.6 (#99368)

joostlek committed:
Use shorthand attributes in iCloud (#99390)

* Use shorthand attributes in iCloud

* Use shorthand attributes in iCloud

piitaya committed:
Add entity component translation for water heater away mode attribute (#99394)

emontnemery committed:
Improve template sensor config flow validation (#99373)

bramkragten committed:
Update frontend to 20230831.0 (#99405)

ludeeus committed:
Add remote alias to connection info response (#99410)

emontnemery committed:
Add documentation URL for homeassistant_yellow (#99336)

* Add documentation URL for homeassistant_yellow

* Fix test

* Tweak

puddly committed:
Initialize ZHA device database before connecting to the radio (#98082)

* Create ZHA entities before attempting to connect to the coordinator

* Delete the ZHA gateway object when unloading the config entry

* Only load ZHA groups if the coordinator device info is known offline

* Do not create a coordinator ZHA device until it is ready

* [WIP] begin fixing unit tests

* [WIP] Fix existing unit tests (one failure left)

* Fix remaining unit test

jpbede committed:
Move tankerkoenig coordinator and base entity to its own file (#99416)

* Move tankerkoenig coordinator and entity to its own file

* Add coordinator.py and entity.py to .coveragerc

bdraco committed:
Significantly reduce overhead to filter event triggers (#99376)

* fast

* cleanups

* cleanups

* cleanups

* comment

* comment

* add more cover

* comment

* pull more examples from forums to validate cover

mib1185 committed:
Address late review for Nextcloud (#99226)

Noltari committed:
Update AEMET-OpenData to v0.4.4 (#99418)

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

kbx81 committed:
Add @kbx81 as esphome codeowner (#99427)

* Add @kbx81 as esphome codeowner

* Add @kbx81 as esphome codeowner, take 2

connorgallopo commented:
Ahhh @teharris1 I just saw you update `pyinsteon` and just made all these same changes locally 🤣 

Great work dude 👍

joostlek committed:
Use shorthand attributes in Isy994 (#99395)

allenporter committed:
Update google-nest-sdm to 3.0.2 (#99175)

* Update google-nest-sdm to 3.0.2

* Fix device typing

* Update homeassistant/components/nest/device_info.py

Co-authored-by: jan iversen <jancasacondor@gmail.com>

---------

Co-authored-by: jan iversen <jancasacondor@gmail.com>

kbickar committed:
Update asynsleepiq library to 1.3.7 (#99431)

jpbede committed:
Move airnow coordinator to its own file (#99423)

joostlek committed:
Use shorthand attributes in Insteon (#99392)

joostlek committed:
Use shorthand attributes in iZone (#99397)

rappenze committed:
Discover more power and energy sensors in fibaro integration (#98253)

puddly committed:
Create a ZHA repair when directly accessing a radio with multi-PAN firmware (#98275)

* Add the SiLabs flasher as a dependency

* Create a repair if the wrong firmware is detected on an EZSP device

* Update homeassistant/components/zha/strings.json

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

* Provide the ZHA config entry as a reusable fixture

* Create a separate repair when using non-Nabu Casa hardware

* Add unit tests

* Drop extraneous `config_entry.add_to_hass` added in 021def44

* Fully unit test all edge cases

* Move `socket://`-ignoring logic into repair function

* Open a repair from ZHA flows when the wrong firmware is running

* Fix existing unit tests

* Link to the flashing section in the documentation

* Reduce repair severity to `ERROR`

* Make issue persistent

* Add unit tests for new radio probing states

* Add unit tests for new config flow steps

* Handle probing failure raising an exception

* Implement review suggestions

* Address review comments

---------

Co-authored-by: c0ffeeca7 <38767475+c0ffeeca7@users.noreply.github.com>

piitaya committed:
Use common key for away mode state translations (#99425)

emontnemery committed:
Fix template helper strings (#99456)

bramkragten committed:
Update frontend to 20230901.0 (#99464)

raman325 committed:
Fix device name in zwave_js repair flow (#99414)

ricmik committed:
Turn off Mill heaters using local API (#99348)

* Update mill_local

* Add ability to turn heater on and off

* Use OperationMode from upstream library

* Fix: compare against value

jpbede committed:
Use snapshot assertion for netatmo diagnostics test (#99159)

bdraco committed:
Avoid linear search of MQTT SUPPORTED_COMPONENTS (#99459)

By making this a set we avoid the linear search in async_discovery_message_received

oandrew committed:
Update aiotractive to 0.5.6 (#99477)

bdraco committed:
Refactor MQTT discovery to avoid creating closure if hash already in discovery_pending_discovered (#99458)

bdraco committed:
Reduce overhead to process and publish MQTT messages (#99457)

Jc2k committed:
Update bluetooth-data-tools to 1.11.0 (#99485)

gjohansson-ST committed:
Fix default language in Workday (#99463)

Workday fix default language

rcloran committed:
Handle timestamp sensors in Prometheus integration (#98001)

jimmyd-be committed:
Fix translation bug Renson sensors (#99461)

* Fix translation bug

* Revert "Fix translation bug"

This reverts commit 84b5e90dac1e75a4c9f6d890865ac42044858682.

* Fixed translation of Renson sensor

jbouwh committed:
Freeze time for MQTT sensor expire tests (#99496)

joostlek committed:
Use shorthand attributes in Hunterdouglas powerview (#99386)

PaarthShah committed:
Change matrix component to use matrix-nio instead of matrix_client (#72797)

Danielhiversen committed:
Update Tibber library to 0.28.2 (#99115)

Jc2k committed:
Support tracking private bluetooth devices (#99465)

Co-authored-by: J. Nick Koston <nick@koston.org>

joostlek committed:
Use shorthand attributes in hlk_sw16 (#99383)

emontnemery committed:
Log unexpected exceptions causing recorder shutdown (#99445)

bdraco commented on homeassistant/components/insteon/light.py:
```diff
@@ -51,6 +52,16 @@ class InsteonDimmerEntity(InsteonEntity, LightEntity):
+    def __init__(self, device: InsteonDevice, group: int) -> None:
+        """Init the InsteonDimmerEntity entity."""
+        super().__init__(device=device, group=group)
+        try:
+            if not self._insteon_device_group.is_dimmable:
+                self._attr_color_mode = ColorMode.ONOFF
+                self._attr_supported_color_modes = {ColorMode.ONOFF}
+        except AttributeError:
+            pass
```
Can we avoid an exception being throw here?

emontnemery committed:
Don't compile missing statistics when running tests (#99446)

bdraco committed:
Switch mqtt to use async_call_later where possible (#99486)

bdraco committed:
Speed up calls to the all states api (#99462)

bdraco committed:
Switch isy994 to use async_call_later (#99487)

async_track_point_in_utc_time is not needed here since we only need to call
at timedelta(hours=25)

jbouwh committed:
Rework on mqtt certificate tests (#99503)

* Shared fixture on TEMP_DIR_NAME mock in MQTT tests

* Improve mqtt certificate file tests

* Update tests/components/mqtt/test_util.py

Co-authored-by: J. Nick Koston <nick@koston.org>

* Update tests/components/mqtt/conftest.py

Co-authored-by: J. Nick Koston <nick@koston.org>

* Avoid blocking code

* typo in sub function

---------

Co-authored-by: J. Nick Koston <nick@koston.org>

allenporter committed:
Code cleanup for nest device info (#99511)

bdraco committed:
Improve lingering timer checks (#99472)

starkillerOG committed:
Check new IP of Reolink camera from DHCP (#99381)

Co-authored-by: J. Nick Koston <nick@koston.org>

bdraco committed:
Bump zeroconf to 0.91.1 (#99490)

starkillerOG committed:
Motion blinds duplication reduction using entity baseclass (#99444)

starkillerOG committed:
Motion blinds add translations (#99078)

bdraco committed:
Bump aiohomekit to 3.0.2 (#99514)

rikroe committed:
Add long-term statistics to BMW sensors (#99506)

* Add long-term statistics to BMW ConnectedDrive sensors

* Add sensor test snapshot

---------

Co-authored-by: rikroe <rikroe@users.noreply.github.com>

bdraco committed:
Reduce overhead to retry config entry setup (#99471)

Jc2k committed:
Add sensors to private_ble_device (#99515)

cdce8p committed:
Fix zha test RuntimeWarning (#99519)

allenporter committed:
Extend template entities with a script section (#96175)

* Extend template entities with a script section

This allows making a trigger entity that triggers a few times a day,
and allows collecting data from a service resopnse which can be
fed into a template entity.

The current alternatives are to publish and subscribe to events or to
store data in input entities.

* Make variables set in actions accessible to templates

* Format code

---------

Co-authored-by: Erik <erik@montnemery.com>

elupus committed:
Bump gardena_bluetooth to 1.4.0 (#99530)

bdraco committed:
Bump zeroconf to 0.93.1 (#99516)

* Bump zeroconf to 0.92.0

changelog: https://github.com/python-zeroconf/python-zeroconf/compare/0.91.1...0.92.0

* drop unused argument

* Update tests/components/thread/test_diagnostics.py

* lint

* again

* bump again since actions failed to release the wheels

cdce8p committed:
Update pytest warning filter (#99521)

Jc2k committed:
Add suggest_display_precision to private_ble_device (#99529)

bdraco committed:
Refactor async_call_at and async_call_later event helpers to avoid creating closures (#99469)

bdraco committed:
Bump bleak to 0.21.0 (#99520)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

mikey0000 committed:
Bugfix: Electric Kiwi reduce interval so oauth doesn't expire (#99489)

decrease interval time as EK have broken/changed their oauth again

yuxincs committed:
Handle gracefully when unloading apcupsd config entries (#99513)

bdraco committed:
Reduce Bluetooth coordinator/processor overhead (#99526)

engrbm87 committed:
Change calculation methods to a fixed list (#99535)

bdraco committed:
Bump bleak-retry-connector to 3.1.2 (#99540)

bdraco committed:
Bump aioesphomeapi to 16.0.4 (#99541)

bdraco committed:
Speed up fetching states by domain (#99467)

mikeodr committed:
Fix recollect_waste month time boundary issue (#99429)

janiversen committed:
Read modbus data before scan_interval (#99243)

Read before scan_interval.

allenporter committed:
Rename nest test_sensor_sdm.py to test_sensor.py (#99512)

janiversen committed:
Modbus switch, allow restore "unknown" (#99533)

engrbm87 committed:
Allow glances entries with same IP but different ports (#99536)

bdraco committed:
Make bond BPUP callback a HassJob (#99470)

janiversen committed:
Make validator for modbus table controlled (#99092)

Kane610 committed:
Bump aiounifi to v60 (#99548)

allenporter committed:
Remove nest legacy service descriptions and translations (#99510)

bdraco committed:
Bump zeroconf to 0.96.0 (#99549)

cdce8p committed:
Fix tolo test warning (#99555)

cdce8p committed:
Fix sql test warning (#99556)

gjohansson-ST committed:
Fix loading filesize coordinator from wrong place (#99547)

* Fix loading filesize coordinator from wrong place

* aboslute in executor

* combine into executor

joostlek committed:
Use shorthand attributes for Heos (#99344)

engrbm87 committed:
Add strict typing to glances (#99537)

joostlek committed:
Move static shorthand devolo attributes outside of constructor (#99234)

Co-authored-by: Guido Schmitz <Shutgun@users.noreply.github.com>

joostlek committed:
Remove unused attributes from Econet (#99242)

joostlek committed:
Use shorthand attributes in Home connect (#99385)

Co-authored-by: Robert Resch <robert@resch.dev>

joostlek committed:
Use shorthand attributes for DLNA dmr (#99236)

rcloran committed:
Enumerate available states in Prometheus startup (#97993)

bdraco committed:
Bump zeroconf to 0.97.0 (#99554)

changelog: https://github.com/python-zeroconf/python-zeroconf/compare/0.96.0...0.97.0

ratsept committed:
Fix battery reading in SOMA API (#99403)

Co-authored-by: Robert Resch <robert@resch.dev>

jpbede committed:
Move london underground coordinator to its own file (#99550)

joostlek committed:
Use shorthand attributes in Laundrify (#99586)

joostlek committed:
Use shorthand attributes in LCN (#99587)

joostlek committed:
Use shorthand attributes in Juicenet (#99575)

joostlek committed:
Use shorthand attributes in Kulersky (#99583)

joostlek committed:
Bump pyenphase to 1.9.1 (#99574)

joostlek committed:
Use shorthand attributes in Keenetic (#99577)

emontnemery committed:
Don't set assumed_state in cover groups (#99391)

emontnemery committed:
Don't set assumed_state in fan groups (#99399)

emontnemery committed:
Set state of entity with invalid state to unknown (#99452)

* Set state of entity with invalid state to unknown

* Add test

* Apply suggestions from code review

Co-authored-by: Robert Resch <robert@resch.dev>

* Update test_entity.py

---------

Co-authored-by: Robert Resch <robert@resch.dev>

emontnemery committed:
Validate state in template helper preview (#99455)

* Validate state in template helper preview

* Deduplicate state validation

emontnemery committed:
Add loader.async_get_loaded_integration (#99440)

* Add loader.async_get_loaded_integration

* Decorate async_get_loaded_integration with @callback

engrbm87 committed:
Use `CONF_SALT` correctly in config_flow validation (#99597)

joostlek committed:
Use shorthand attributes in Lyric (#99593)

joostlek committed:
Use shorthand attributes in Motion eye (#99596)

joostlek committed:
Use shorthand attributes in Nanoleaf (#99601)

puddly committed:
Bump ZHA dependencies (#99561)

joostlek committed:
Remove unneeded name property from Logi Circle (#99604)

joostlek committed:
Use shorthand attributes in Mutesync (#99600)

DCSBL committed:
Bumb python-homewizard-energy to 2.1.0 (#99598)

engrbm87 committed:
Bump simplepush api to 2.2.3 (#99599)

joostlek committed:
Remove default state from Nibe (#99611)

Remove start state

joostlek committed:
Use shorthand attributes in Nest (#99606)

joostlek committed:
Use shorthand attributes in Octoprint (#99623)

joostlek committed:
Remove slugify from tomorrowio unique id (#99006)

dknowles2 committed:
Bump pyschlage to 2023.9.0 (#99624)

emontnemery committed:
Revert "Deprecate timer start optional duration parameter" (#99613)

Revert "Deprecate timer start optional duration parameter (#93471)"

This reverts commit 2ce5b08fc36e77a2594a39040e5440d2ca01dff8.

emontnemery committed:
Small cleanup of WS command render_template (#99562)

bdraco committed:
Fix module check in _async_get_flow_handler (#99509)

We should have been checking for the module in hass.data[DATA_COMPONENTS]
and not hass.config.components as the check was ineffective if there were
no existing integrations instances for the domain which is the case for
discovery or when the integration is ignored

mib1185 committed:
Mark AVM Fritz!Smarthome as Gold integration (#97086)

set quality scale to gold

bramkragten committed:
Update frontend to 20230904.0 (#99636)

joostlek committed:
Move variables out of constructor in Nobo hub (#99617)

emontnemery committed:
Small cleanup of TemplateEnvironment (#99571)

* Small cleanup of TemplateEnvironment

* Fix typo

gjohansson-ST committed:
Fix missing unique id in SQL (#99641)

joostlek committed:
Move variables out of constructor in nightscout (#99612)

* Move variables out of constructor in nightscout

* Update homeassistant/components/nightscout/sensor.py

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

---------

Co-authored-by: G Johansson <goran.johansson@shiftit.se>

gjohansson-ST committed:
Fix not stripping no device class in template helper binary sensor (#99640)

Strip none template helper binary sensor

bdraco committed:
Speed up responding to states being polled via API (#99621)

* Speed up responding to states being polled via API

Switch to using `as_dict_json` to avoid serializing states over and over when the
states api is polled since the mobile app is already building the cache as it also
polls the states via the websocket_api

* Speed up responding to states being polled via API

Switch to using `as_dict_json` to avoid serializing states over and over when the
states api is polled since the mobile app is already building the cache as it also
polls the states via the websocket_api

* fix json

* cover

MartinHjelmare committed:
Move non legacy stt models out from legacy module (#99582)

bdraco committed:
Fix Bluetooth passive update processor dispatching updates to unchanged entities (#99527)

* Fix passive update processor dispatching updates to unchanged entities

* adjust tests

* coverage

* fix

* Update homeassistant/components/bluetooth/update_coordinator.py

bdraco committed:
Improve performance of google assistant supported checks (#99454)

* Improve performance of google assistant supported checks

* tweak

* tweak

* split function

* tweak

bdraco committed:
Fix mobile app dispatcher performance (#99647)

Fix mobile app thundering heard

The mobile_app would setup a dispatcher to listener for updates on
every entity and reject the ones that were not for the unique id
that it was intrested in.

Instead we now register for a signal per unique id since we were
previously generating O(entities*sensors*devices) callbacks which
was causing the event loop to stall when there were a large
number of mobile app users.

joostlek committed:
Use shorthand attributes for Picnic (#99633)

dependabot[bot] committed:
Bump actions/checkout from 3.6.0 to 4.0.0 (#99651)

joostlek committed:
Use shorthand attributes in Nuheat (#99618)

Noltari committed:
Update aioairzone to v0.6.8 (#99644)

* Update aioairzone to v0.6.8

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

* Trigger CI

---------

Signed-off-by: Álvaro Fernández Rojas <noltari@gmail.com>

itpeters committed:
Fix long press event for matter generic switch (#99645)

joostlek committed:
Use shorthand attributes in Openhome (#99629)

joostlek committed:
Use shorthand attributes in Kodi (#99578)

joostlek committed:
Use shorthand attributes in Konnected (#99580)

joostlek committed:
Use shorthand attributes in Kostal Plenticore (#99581)

joostlek committed:
Use shorthand attributes in Logi circle (#99592)

joostlek committed:
Use shorthand attributes in NZBGet (#99622)

joostlek committed:
Use shorthand attributes in Omnilogic (#99626)

joostlek committed:
Use shorthand attributes in Ondilo ico (#99627)

raman325 committed:
Bump zwave-js-server-python to 0.51.1 (#99652)

* Bump zwave-js-server-python to 0.51.1

* Update test

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

joostlek committed:
Use shorthand attributes in Neato (#99605)

Co-authored-by: Robert Resch <robert@resch.dev>

bdraco committed:
Replace lambda with attrgetter in hassfest (#99662)

bdraco committed:
Replace lambda with attrgetter in device_tracker device_trigger (#99663)

bdraco committed:
Replace lambda with itemgetter in script/gen_requirements_all.py (#99661)

bdraco committed:
Replace lambda with attrgetter in homekit_controller (#99666)

joostlek committed:
Use shorthand attributes for Plaato (#99634)

Co-authored-by: Robert Resch <robert@resch.dev>

bramkragten committed:
Update frontend to 20230905.0 (#99677)

puddly committed:
Fix ZHA startup creating entities with non-unique IDs (#99679)

* Make the ZHAGateway initialization restartable so entities are unique

* Add a unit test

Kane610 committed:
Bump aiounifi to v61 (#99686)

* Bump aiounifi to v61

* Alter a test to cover the upstream change

Danielhiversen committed:
Bump millheater to 0.11.2 (#99683)

Update Mill lib

Signed-off-by: Daniel Hjelseth Høyer <github@dahoiv.net>

killer0071234 committed:
Bump zamg to 0.3.0 (#99685)

starkillerOG committed:
Bump reolink_aio to 0.7.9 (#99680)

jbouwh committed:
Remove imap_email_content integration (#99484)

bdraco committed:
Replace lambda in script/gen_requirements_all.py with str.lower (#99665)

bdraco committed:
Switch lambda to attrgetter in zha (#99660)

bdraco committed:
Refactor dispatcher to reduce run time and memory overhead (#99676)

* Fix memory leak in dispatcher removal

When we removed the last job/callable from the dict for the
signal we did not remove the dict for the signal which meant
it leaked

* comment

* cleanup a bit more

bdraco committed:
Switch homekit config flow sorted to use itemgetter (#99658)

Avoids unnecessary lambda

bdraco committed:
Bump aioesphomeapi to 16.0.5 (#99698)

changelog: https://github.com/esphome/aioesphomeapi/compare/v16.0.4...v16.0.5

fixes `RuntimeError: set changed size during iteration`

https://github.com/esphome/aioesphomeapi/pull/538

some added debug logging which may help with https://github.com/home-assistant/core/issues/98221

joostlek committed:
Add codeowner for Withings (#99681)

bieniu committed:
Show OTA update progress for Shelly gen2 devices (#99534)

* Show OTA update progress

* Use an event listener instead of a dispatcher

* Add tests

* Fix name

* Improve tests coverage

* Fix subscribe/unsubscribe logic

* Use async_on_remove()

engrbm87 committed:
Display channel number in Bravia TV if title is not available (#99567)

Display channel number if title is not available

tronikos committed:
Add a comment why state_class=total (#99703)

jpbede committed:
Move template coordinator to its own file (#99419)

* Move template update coordinator to its own file

* Add coordinator.py to .coveragerc

* Remove coordinator.py to .coveragerc

* Apply suggestions from code review

* Update homeassistant/components/template/coordinator.py

* Copy over fixes from upstream

---------

Co-authored-by: Erik Montnemery <erik@montnemery.com>
Co-authored-by: G Johansson <goran.johansson@shiftit.se>

emontnemery committed:
Include template listener info in template preview (#99669)

jpbede committed:
Use shorthand attributes in Trend (#99695)

emontnemery committed:
Allow specifying a custom log function for template render (#99572)

* Allow specifying a custom log function for template render

* Bypass template cache when reporting errors + fix tests

* Send errors as events

* Fix logic for creating new TemplateEnvironment

* Add strict mode back

* Only send error events if report_errors is True

* Force test of websocket_api only

* Debug test

* Run pytest with higher verbosity

* Timeout after 1 minute, enable syslog output

* Adjust timeout

* Add debug logs

* Fix unsafe call to WebSocketHandler._send_message

* Remove debug code

* Improve test coverage

* Revert accidental change

* Include severity in error events

* Remove redundant information from error events

emontnemery committed:
Fix unit conversion for gas cost sensor (#99708)

bramkragten committed:
Update frontend to 20230906.0 (#99715)

c0ffeeca7 committed:
Yellow LED controls: rename LEDs (#99710)

- reorder, to reflect placement on board, left to right (yellow, green, red)

bdraco committed:
Use loop time to set context (#99701)

* Use loop time to set context

loop time is faster than utcnow, and since its only used internally it can
be switched without a breaking change

* fix mocking

cdce8p committed:
Fix pylint plugin test DeprecationWarning (#99711)

jbouwh committed:
Revert "Remove imap_email_content integration" (#99713)

jbouwh committed:
Postpone Imap_email_content removal (#99721)

starkillerOG committed:
Reolink onvif not supported fix (#99714)

* only subscibe to ONVIF if supported

* Catch NotSupportedError when ONVIF is not supported

* fix styling

bramkragten committed:
Update frontend to 20230906.1 (#99733)

dknowles2 committed:
Add binary_sensor to Schlage (#99637)

* Add binary_sensor to Schlage

* Apply suggestions from code review

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

---------

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>

emontnemery committed:
Correct state attributes in group helper preview (#99723)

emontnemery committed:
Don't allow changing device class in template binary sensor options (#99720)

emontnemery committed:
Correct state attributes in template helper preview (#99722)

emontnemery committed:
Send template render errors to template helper preview (#99716)

dknowles2 committed:
Fix the Hydrawise status sensor (#99271)

cdce8p committed:
Fix asyncio.wait typing (#99726)

cdce8p committed:
Fix tradfri asyncio.wait (#99730)

cdce8p committed:
Update pre-commit to 3.4.0 (#99737)

cdce8p committed:
Improve blink config_flow typing (#99579)

joostlek committed:
Revert "Bump pyoverkiz to 1.10.1 (#97916)" (#99742)

cdce8p committed:
Update pytest-aiohttp to 1.0.5 (#99744)

loopj committed:
Enable strict typing for Climate component (#99301)

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

mkmer committed:
Raise error on open/close failure in Aladdin Connect (#99746)

Raise error on open/close failure

Vantomas committed:
Add additional fields for 3-phase UPS to nut (#98625)

Co-authored-by: J. Nick Koston <nick@koston.org>

jpbede committed:
Improve typing of trend component (#99719)

* Some typing in trend component

* Add missing type hint

* Enable strict typing on trend

jbouwh committed:
Handle alexa invalid climate temp adjustment (#99740)

* Handle temp adjust when target state not set

* Update homeassistant/components/alexa/errors.py

Co-authored-by: Robert Resch <robert@resch.dev>

* black

---------

Co-authored-by: Robert Resch <robert@resch.dev>

bdraco committed:
Bump zeroconf to 0.98.0 (#99748)

bdraco committed:
Bump dbus-fast to 1.95.0 (#99749)

joostlek committed:
Remove deprecated entities from OpenTherm Gateway (#99712)

joostlek committed:
Use shorthand attributes in Progetti  (#99772)

Use shorthand attributes in Progetti shorthand

joostlek committed:
Use shorthand attributes in Plum (#99770)

Use shorthand attributes in Plum shorthand

bachya committed:
Bump `aiorecollect` to 2023.09.0 (#99780)

bdraco committed:
Bump sense_energy to 0.12.1 (#99763)

bieniu committed:
Add energy meter sensors for Shelly Pro EM (#99747)

* Add support for Pro EM

* Improve get_rpc_channel_name()

* Revert an unintended change

* Add tests

Quentame committed:
Fix Freebox disk free space sensor (#99757)

* Fix Freebox disk free space sensor

* Add initial value assert to check results

LaStrada committed:
Add my self as codeowner for airthings_ble (#99799)

Update airthings_ble codeowner

ArdaSeremet committed:
Bump ProgettiHWSW to 0.1.3 (#92668)

* Update manifest.json

* Update requirements_test_all.txt

* Update requirements_all.txt

* Updated dependencies file.

* Update manifest.json with correct naming convention.

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Updated requirements.

---------

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

emontnemery committed:
Always set severity level flag on render_template error events (#99804)

engrbm87 committed:
Add strict typing to islamic prayer times (#99585)

* Add strict typing to islamic prayer times

* fix mypy errors

bdraco committed:
Bump pyenphase to 1.9.3 (#99787)

* Bump pyenphase to 1.9.2

changelog: https://github.com/pyenphase/pyenphase/compare/v1.9.1...v1.9.2

Handle the case where the user has manually specified a password for
local auth with firmware < 7.x but its incorrect.

The integration previously accepted any wrong password and would
reduce functionality down to what works without a password. We now
preserve that behavior to avoid breaking existing installs.

* bump

swamplynx committed:
Bump pylutron-caseta to v0.18.2 (#99789)

* Bump pylutron-caseta to v0.18.2

Minor bump to pylutron-caseta requirement to support wall mounted occupancy sensor device type in latest RA3 firmware.

* Update requirements_all.txt for pylutron-caseta 0.18.2

* Update requirements_test_all.txt for pylutron-caseta 0.18.2

elmurato committed:
Use config entry ID as unique ID and remove dependency to getmac in Minecraft Server (#97837)

* Use config entry ID as unique ID

* Add entry migration to v2 and and remove helper module

* Remove unneeded strings

* Add asserts for config, device and entity entries and improve comments

* Add debug log for config entry migration

* Reset config entry unique ID and use config entry ID instead

* Remove unnecessary unique ID debug log

* Revert usage of constants for tranlation keys and use dash as delimiter for entity unique id suffix

* Revert "Revert usage of constants for tranlation keys and use dash as delimiter for entity unique id suffix"

This reverts commit 07de334606054097e914404da04950e952bef6d2.

* Remove unused logger in entity module

pszafer committed:
Add support for more busy codes for Epson (#99771)

add support for more busy codes

joostlek committed:
Use correct config entry id in Livisi (#99812)

emontnemery committed:
Fix typo in TrackTemplateResultInfo (#99809)

emontnemery committed:
Improve error handling in /api/states POST (#99810)

joostlek committed:
Use shorthand attributes in Hisense (#99355)

joostlek committed:
Use shorthand attributes in Plex (#99769)

Co-authored-by: Robert Resch <robert@resch.dev>

cdce8p committed:
Update coverage to 7.3.1 (#99805)

jrieger committed:
Change AVM FRITZ!Box Call monitor sensor into an enum (#99762)

Co-authored-by: Joost Lekkerkerker <joostlek@outlook.com>
Co-authored-by: Marc Mueller <30130371+cdce8p@users.noreply.github.com>

joostlek committed:
Remove unused variable from rainbird (#99824)

joostlek committed:
Use shorthand attributes in Sense (#99833)

joostlek committed:
Use shorthand attributes in Roomba (#99831)

joostlek committed:
Use shorthand attributes in SharkIQ (#99836)

joostlek committed:
Use shorthand attributes in Rachio (#99823)

joostlek committed:
Use shorthand attributes in Snapcast (#99840)

joostlek committed:
Move shorthand attributes out of constructor in Sensibo (#99834)

Use shorthand attributes in Sensibo

Quentame committed:
Fix Freebox Home battery sensor (#99756)

joostlek committed:
Move unit of temperature to descriptions in Sensibo (#99835)

joostlek committed:
Use shorthand attributes in Opentherm gateway (#99630)

puddly committed:
Bump ZHA dependencies (#99855)

joostlek committed:
Use shorthand attributes in Sonarr (#99844)

joostlek committed:
Move shorthand attributes out of Snooz constructor (#99842)

joostlek committed:
Use shorthand attributes in Ring (#99829)

joostlek committed:
Use shorthand attributes in Rainforest eagle (#99825)

joostlek committed:
Use str instead of string placeholders in solaredge (#99843)

bdraco committed:
Bump zeroconf to 0.99.0 (#99853)

bdraco committed:
Bump dbus-fast to 1.95.2 (#99852)

gjohansson-ST committed:
Add device class and UoM in Sensibo Number entities (#99861)

* device class and uom number platform

* icons

jimmyd-be committed:
Renson fan (#94495)

* Add fan feature

* Changed order of platform

* Use super()._handle_coordinator_update()

* format file

* Set _attr_has_entity_name

* Cleanup Fan code

* Refresh after setting ventilation speed + translation

* remove unused translation key

janiversen committed:
Modbus scale parameter cuts decimals (#99758)

mkmer committed:
Add Diagnostic platform to Aladdin Connect (#99682)

* Add diagnostics platform

* Add diagnostic platform

* Add raw data to diagnostics

* Remove config data
bump aioaladdinconnect, use new doors property for diag

* remove unnecessary component config
refactor diag output

teharris1 commented:
Damn it. Rebasing broke me again.
